### ISSUE TITLE
agent-auth ERROR: SSL error 

### ISSUE THREAD

**Issue by alberpilot** [wazuh/wazuh#2905]
Hello team

|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.8.2 | 3814 |Agent | Packages |

If you use by mistake the manager remoted port instead the authd port we have the following message: 
```
[root@agent vagrant]# /var/ossec/bin/agent-auth -m 172.16.1.2 -p 1514 -P TopSecret
2019/03/25 18:18:04 agent-auth: INFO: Started (pid: 5408).
2019/03/25 18:18:04 agent-auth: ERROR: SSL error (-1). Exiting.
```
I think that the error is not descriptive and it invites the user to review openssl version, problems related with openssl package, etc. 

Best regards,

**Comment by juan70** [comment]
2019-03-26
----------
- Trying to reproduce the error, but no success. The only way has been to remove a piece of code.

**Comment by juan70** [comment]
When the vagrant box is being provisioned:
```
==> agent: Running provisioner: shell...
    agent: Running: /tmp/vagrant-shell20190326-1128-5kq8io.sh
    agent: Selecting previously unselected package wazuh-agent.
    agent: (Reading database ... 58167 files and directories currently installed.)
    agent: Preparing to unpack .../wazuh-agent_3.9.0-15032019_amd64.deb ...
    agent: Unpacking wazuh-agent (3.9.0-15032019) ...
    agent: Setting up wazuh-agent (3.9.0-15032019) ...
    agent: Processing triggers for systemd (229-4ubuntu21.15) ...
    agent: Processing triggers for ureadahead (0.100.0-19) ...
    agent: 2019/03/26 16:03:09
    agent: agent-auth:
    agent: INFO:
    agent: Started (pid: 19177).
    agent: 2019/03/26 16:03:09
    agent: agent-auth:
    agent: ERROR:
    agent: SSL error (-1). Exiting.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
When registering the agent by hand (after `vagrant ssh agent` and `sudo su -`):
```
root@agent:~# /var/ossec/bin/agent-auth -m 172.16.1.2 -p 1514 -P TopSecret
2019/03/26 16:04:56 agent-auth: INFO: Started (pid: 19272).
2019/03/26 16:04:56 agent-auth: ERROR: Unable to connect to 172.16.1.2:1514
```
These messages are thrown from `src/os_auth/main-client.c`:
```
329     /* Connect via TCP */
330     sock = OS_ConnectTCP(port, ipaddress, 0);
331     if (sock <= 0) {
332         merror("Unable to connect to %s:%d", ipaddress, port);
333         free(buf);
334         exit(1);
335     }
336
337     /* Connect the SSL socket */
338     ssl = SSL_new(ctx);
339     sbio = BIO_new_socket(sock, BIO_NOCLOSE);
340     SSL_set_bio(ssl, sbio, sbio);
341
342     ret = SSL_connect(ssl);
343     if (ret <= 0) {
344         ERR_print_errors_fp(stderr);
345         merror("SSL error (%d). Exiting.", ret);
346         free(buf);
347         exit(1);
348     }
```

**Comment by juan70** [comment]
Description of how the testing VM is provisioned:
- manager's `ossec.conf`
  - remote section
    - change protocol to tcp
    - change port 1514 to 1524
  - auth section
    - change port 1515 to 1525
    - use password
  - set the password in `/var/ossec/etc/authd.pass`
  - restart manager

- agent
  - just install the wazuh-agent package
  - try to register with the correct password and on port 1524 (which is now remoted, not authd)
  => the error is thrown

Interesting note for debugging: if the agent machine is **not** provisioned and the package is installed **after** the VM is up and running, trying to register **always** throws the error, at each intent.

As a matter of fact, installing manager and agent from sources on other VMs and setting up the manager as described above allows to reproduce the error.

**Comment by juan70** [comment]
Adding a few debugging messages in the source code, we get:
```
# /var/ossec/bin/agent-auth -m 192.168.56.113 -p 1524 -P TopSecret
2019/03/27 11:38:32 agent-auth: INFO: Started (pid: 13270).
>>> INFO: Connect via TCP.
>>> INFO: Connect via TCP passed. sock = >3<.
>>> INFO: SSL_connect.
2019/03/27 11:38:32 agent-auth: ERROR: SSL error (-1). Exiting.
2019/03/27 11:38:32 agent-auth: ERROR: >>> errno >0<.
2019/03/27 11:38:32 agent-auth: ERROR: >>> ERR_get_error >0<.
2019/03/27 11:38:32 agent-auth: ERROR: >>> ERR_error_string >error:00000000:lib(0):func(0):reason(0)<.
2019/03/27 11:38:32 agent-auth: ERROR: >>> ERR_reason_error_string >(null)<.
2019/03/27 11:38:32 agent-auth: ERROR: >>> SSL_get_error >5<.
>>> SSL_ERROR_SYSCALL.
```
There seems to be no easy way to make a useful diagnose here. The TCP connection is established, but the SSL connection fails. `errno` is `0`, and the most *descriptive* message is a generic `SSL_ERROR_SYSCALL` when `SSL_get_error` returns `5`.

The easiest solution is to change the error message to include a suggestion that the port may be incorrect, along with a *translation* of the value returned by `SSL_connect()` into the SSL error code through `SSL_get_error()`.

**Comment by chemamartinez** [comment]
Merged: #2941
